ReactAndroid/src/main/java/com/facebook/react/bridge/JSBundleLoader.java

```diff
@@ -22,18 +22,22 @@
    * should be used. JS bundle will be read from assets directory in native code to save on passing
    * large strings from java to native memory.
    */
-  public static JSBundleLoader createAssetLoader(
-      final AssetManager assetManager,
-      final String assetFileName) {
+  public static JSBundleLoader createFileLoader(
+      final Context context,
+      final String fileName) {
     return new JSBundleLoader() {
       @Override
       public void loadScript(ReactBridge bridge) {
-        bridge.loadScriptFromAssets(assetManager, assetFileName);
+        if (fileName.startsWith("assets://")) {
+          bridge.loadScriptFromAssets(context.getAssets(), fileName.replaceFirst("assets://", ""));
+        } else {
+          bridge.loadScriptFromFile(fileName, fileName);
+        }
       }
 
       @Override
       public String getSourceUrl() {
-        return "file:///android_asset/" + assetFileName;
+        return fileName;
```

Comment:
@kmagiera I guess you mean JS, not JSC. Yes, this is only used so JS knows where the JS bundle was loaded from (local file on device/emulator or network). JS uses this to determine where to load assets from (drawables in the APK or from the dev server).